src/carnot/exec/otel_export_sink_node_test.cc

```diff
@@ -146,6 +146,69 @@ metrics {
   EXPECT_EQ(url_, "otlp.px.dev");
 }
 
+TEST_F(OTelExportSinkNodeTest, non_utf_8) {
+  std::string operator_pb_txt = R"(
+endpoint_config {
+  url: "otlp.px.dev"
+    headers {
+      key: "api_key"
+      value: "abcd"
+    }
+}
+metrics {
+  name: "http.resp.latency"
+  time_column_index: 0
+  gauge { int_column_index: 1 }
+  attributes {
+    name: "test"
+    column {
+      column_type: STRING
+      column_index: 2
+    }
+  }
+})";
+  planpb::OTelExportSinkOperator otel_sink_op;
+
+  EXPECT_TRUE(google::protobuf::TextFormat::ParseFromString(operator_pb_txt, &otel_sink_op));
+  auto plan_node = std::make_unique<plan::OTelExportSinkOperator>(1);
+  auto s = plan_node->Init(otel_sink_op);
+  RowDescriptor input_rd({types::TIME64NS, types::FLOAT64, types::STRING});
+  RowDescriptor output_rd({});
+
+  std::shared_ptr<const grpc::AuthContext> auth_context;
+
+  std::vector<otelmetricscollector::ExportMetricsServiceRequest> actual_protos(1);
+  size_t i = 0;
+  EXPECT_CALL(*metrics_mock_, Export(_, _, _))
+      .Times(1)
+      .WillRepeatedly(Invoke(
+          [&i, &actual_protos, &auth_context](const auto& ctx, const auto& proto, const auto&) {
+            auth_context = ctx->auth_context();
+            actual_protos[i] = proto;
+            ++i;
+            return grpc::Status::OK;
+          }));
+
+  auto tester = exec::ExecNodeTester<OTelExportSinkNode, plan::OTelExportSinkOperator>(
+      *plan_node, output_rd, {input_rd}, exec_state_.get());
+  char non_utf_8_bytes[] = {static_cast<char>(0xC0)};
```

Comment:
I think if you use an `std::string` here instead of `char[]`, it will avoid the asan error